include/boost/detail/winapi/dll.hpp

```diff
@@ -18,83 +19,186 @@
+    DWORD_      State;
+    DWORD_      Protect;
+    DWORD_      Type;
+    DWORD_      __alignment2;
+};
+
+struct MEMORY_BASIC_INFORMATION32_ {
+    DWORD_ BaseAddress;
+    DWORD_ AllocationBase;
+    DWORD_ AllocationProtect;
+    DWORD_ RegionSize;
+    DWORD_ State;
+    DWORD_ Protect;
+    DWORD_ Type;
+};
+
 #ifdef _WIN64
 typedef INT_PTR_ (WINAPI *FARPROC_)();
 typedef INT_PTR_ (WINAPI *NEARPROC_)();
 typedef INT_PTR_ (WINAPI *PROC_)();
+typedef MEMORY_BASIC_INFORMATION64_ MEMORY_BASIC_INFORMATION_;
 #else
 typedef int (WINAPI *FARPROC_)();
 typedef int (WINAPI *NEARPROC_)();
 typedef int (WINAPI *PROC_)();
+typedef MEMORY_BASIC_INFORMATION32_ MEMORY_BASIC_INFORMATION_;
 #endif // _WIN64
-}}}
+
+const DWORD_ DONT_RESOLVE_DLL_REFERENCES_           = 0x00000001;
+const DWORD_ LOAD_IGNORE_CODE_AUTHZ_LEVEL_          = 0x00000010;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_              = 0x00000002;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE_    = 0x00000040;
+const DWORD_ LOAD_LIBRARY_AS_IMAGE_RESOURCE_        = 0x00000020;
+const DWORD_ LOAD_WITH_ALTERED_SEARCH_PATH_         = 0x00000008;
+}}} // namespace boost::detail::winapi
 
 #if !defined( BOOST_NO_ANSI_APIS )
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 LoadLibraryA(boost::detail::winapi::LPCSTR_ lpFileName);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
+LoadLibraryExA(
+    boost::detail::winapi::LPCSTR_ lpFileName,
+    boost::detail::winapi::HANDLE_ hFile,
+    boost::detail::winapi::DWORD_ dwFlags
+);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 GetModuleHandleA(boost::detail::winapi::LPCSTR_ lpFileName);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::DWORD_ WINAPI
+GetModuleFileNameA(
+    boost::detail::winapi::HMODULE_ hModule,
+    boost::detail::winapi::LPSTR_ lpFilename,
+    boost::detail::winapi::DWORD_ nSize
+);
 #endif
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 LoadLibraryW(boost::detail::winapi::LPCWSTR_ lpFileName);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
+LoadLibraryExW(
+    boost::detail::winapi::LPCWSTR_ lpFileName,
+    boost::detail::winapi::HANDLE_ hFile,
+    boost::detail::winapi::DWORD_ dwFlags
+);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 GetModuleHandleW(boost::detail::winapi::LPCWSTR_ lpFileName);
 
+BOOST_SYMBOL_IMPORT boost::detail::winapi::DWORD_ WINAPI
+GetModuleFileNameW(
+    boost::detail::winapi::HMODULE_ hModule,
+    boost::detail::winapi::LPWSTR_ lpFilename,
+    boost::detail::winapi::DWORD_ nSize
+);
+
+
 BOOST_SYMBOL_IMPORT boost::detail::winapi::BOOL_ WINAPI
 FreeLibrary(boost::detail::winapi::HMODULE_ hModule);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
 GetProcAddress(boost::detail::winapi::HMODULE_ hModule, boost::detail::winapi::LPCSTR_ lpProcName);
-}
-#endif
 
-namespace boost {
-namespace detail {
-namespace winapi {
+BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
+VirtualQuery(
```

Comment:
This function is most probably declared incorrectly (I haven't compared it to the actual declaration in Windows SDK, but fairly sure still). The function must use the same types in the signature as it does in Windows SDK, and it surely doesn't reference `boost::detail::winapi::MEMORY_BASIC_INFORMATION32_` or `boost::detail::winapi::MEMORY_BASIC_INFORMATION64_` there. This header should forward-declare the structure/union/whatever `MEMORY_BASIC_INFORMATION` exactly as it is presented in Windows SDK and use a pointer to it in the `VirtualQuery` forward declaration. `boost::detail::winapi::VirtualQuery` should be an inline function that does a `reinterpret_cast` of the `boost::detail::winapi::MEMORY_BASIC_INFORMATION_` pointer to the pointer to the real `MEMORY_BASIC_INFORMATION`.